spring-shell-core/src/main/java/org/springframework/shell/ConfigurableCommandRegistry.java

```diff
@@ -58,6 +58,16 @@ else if (mim == InteractionMode.NONINTERACTIVE) {
 				.collect(Collectors.toMap(e -> e.getKey(), e -> e.getValue()));
 	}
 
+	@Override
+	public void addCommand(String name, MethodTarget target) {
+		commands.put(name, target);
```

Comment:
Should we log warning when a command is overrwrritten? Not sure when this would happen, but it may be helpful to know when/if it does.